packages/compiler/src/core/formatter-fs.ts

```diff
@@ -13,12 +13,14 @@ export interface TypeSpecFormatOptions {
 /**
  * Format all the TypeSpec files.
  * @param patterns List of wildcard pattern searching for TypeSpec files.
+ * @returns list of files which failed to format.
  */
 export async function formatTypeSpecFiles(
```

Comment:
A more consistent api with others would be to return a list of diagnostics. And maybe don't log the error in the formatTypeSpecFiles and let the consumer to log the diagnostics

Reply:
Fair point, I'll give that a shot.  What do you think about changing the return type signature of this existing function?  Should I create a separate function for this new behavior?

Reply:
It goes from void to returning something so I shouldn't be a breaking change. I think it is fine to return `Promise<readonly Diagnostic[]>` or just a boolean showing the success. Just feels like returning a list of files might be confusing that it could be the list of formatted files.

Which actually maybe we should do that:

```ts
interface FormattingResult {
  formattedFiles: string[]
}
export async function formatTypeSpecFiles(): [FormattingResult, readonly Diagnostic[]]

// Or just take a list of `string[]` just depends if we see more data being returned in the future I think.
export async function formatTypeSpecFiles(): [string[], readonly Diagnostic[]]
```

Reply:
Yeah, I'd prefer to have a result object for future use.  Thanks!

Reply:
Pushed a fix for this, let me know what you think!